webapp/channels/src/components/cloud_preview_modal/preview_modal_controller.test.tsx

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {render, screen, fireEvent} from '@testing-library/react';
+import React from 'react';
+import {IntlProvider} from 'react-intl';
+
+import type {PreviewModalContentData} from './preview_modal_content_data';
+import PreviewModalController from './preview_modal_controller';
+
+// Mock the GenericModal component
+jest.mock('@mattermost/components', () => ({
+    GenericModal: ({children, show, onHide, showCloseButton, footerContent}: {
+        children: React.ReactNode;
+        show: boolean;
+        onHide?: () => void;
+        showCloseButton?: boolean;
+        footerContent?: React.ReactNode;
+    }) => (
+        show ? (
+            <div data-testid='modal'>
+                {showCloseButton && (
+                    <button
+                        aria-label='Close'
+                        onClick={onHide}
+                        data-testid='close-button'
+                    >
+                        <span data-testid='close-icon'>{'24'}</span>
+                    </button>
+                )}
+                <div>{children}</div>
+                {footerContent && <div data-testid='modal-footer'>{footerContent}</div>}
+            </div>
+        ) : null
+    ),
+}));
+
+// Mock the icons
+jest.mock('@mattermost/compass-icons/components', () => ({
+    ArrowLeftIcon: ({size}: {size: number}) => <span data-testid='arrow-left-icon'>{size}</span>,
+    ArrowRightIcon: ({size}: {size: number}) => <span data-testid='arrow-right-icon'>{size}</span>,
+    CloseIcon: ({size}: {size: number}) => <span data-testid='close-icon'>{size}</span>,
+}));
+
+describe('PreviewModalController', () => {
+    const mockOnClose = jest.fn();
+
+    const contentData: PreviewModalContentData[] = [
+        {
+            title: {
+                id: 'test.slide1.title',
+                defaultMessage: 'First Slide',
+            },
+            subtitle: {
+                id: 'test.slide1.subtitle',
+                defaultMessage: 'First subtitle',
+            },
+            skuLabel: {
+                id: 'test.slide1.sku_label',
+                defaultMessage: 'First sku label',
+            },
+            videoUrl: 'https://www.youtube.com/watch?v=E3EGLxgNxNA',
```

Comment:
Would it be possible to use a dummy URL for this? I don't know if Testing Library actually would render iframes, but it seems like it would be best to use something locally hosted to avoid having the tests be fragile

Reply:
Since we're not actually using any YouTube videos right now and we can't actually dynamically set these until a future code change, I've opted to just remove the YouTube video support for now. If we do end up needing it in the future, I can add it back at that point.